vault/logical_system.go

```diff
@@ -4273,21 +4275,31 @@ func (b *SystemBackend) pathInternalUIMountsRead(ctx context.Context, req *logic
 			continue
 		}
 
-		if ns.ID == entry.NamespaceID && hasAccess(ctx, entry) {
-			if isAuthed {
-				// If this is an authed request return all the mount info
-				authMounts[entry.Path] = b.mountInfo(ctx, entry)
-			} else {
-				authMounts[entry.Path] = map[string]interface{}{
-					"type":        entry.Type,
-					"description": entry.Description,
-					"options":     entry.Options,
+		if ns.ID == entry.NamespaceID {
+
+			// filters UI login method type dropdown when a user is unauthenticated
+			if entry.Config.ListingVisibility == ListingVisibilityDropdown {
+				authTypes = append(authTypes, entry.Type)
+			}
+
+			if hasAccess(ctx, entry) {
+				if isAuthed {
+					// If this is an authed request return all the mount info
+					authMounts[entry.Path] = b.mountInfo(ctx, entry)
+				} else {
+					authMounts[entry.Path] = map[string]interface{}{
+						"type":        entry.Type,
+						"description": entry.Description,
+						"options":     entry.Options,
+					}
 				}
 			}
 		}
 	}
 	b.Core.authLock.RUnlock()
 
+	authTypes = strutil.RemoveDuplicates(authTypes, false)
```

Comment:
Does the frontend code care about ordering?  Should we sort?

Reply:
Not particularly, we're just looking for if a method exists in the array or not. But on that note, did I misread the function? I thought it sorted already [here](https://github.com/hashicorp/go-secure-stdlib/blob/main/strutil/strutil.go#L251) 

Auths are already lowercased, but probably best to not to assume that 😅

Reply:
Ah, so it does.